src/components/fields/date-field/date-field.tsx

```diff
@@ -96,9 +104,38 @@ export const DateField = (props: IGenericFieldProps<IDateFieldSchema>) => {
 						const localDate = DateTimeHelper.toLocalDateOrTime(value, dateFormat, "date");
 						return !localDate?.isAfter(maxDate);
 					}
+				)
+				.test(
+					"excluded-dates",
+					excludedDatesRule?.["errorMessage"] || ERROR_MESSAGES.DATE.DISABLED_DATES,
+					(value) => {
+						if (!isValidDate(value) || !excludedDatesRule) return true;
```

Comment:
I don't recall the outcome as well but I think the idea is it is not meant to accept the disabled dates in any case.

the onus will be on the dev who needs `allowDisabledSelection` to do the additional handling. e.g. custom validation, conditional validation, etc

Reply:
I think we need to skip the validation if `allowDisabledSelection` is true here, otherwise the validation within FEE will always fail for those disallowed dates, so they can't really do custom validation on top?

Reply:
true that yet I am concerned it can be confusing, the validation rules are meant to validate the values and prevent submission.

one alternative will be to set soft validation, a warning will appear instead of an error and the submission will not be blocked.

Reply:
oh yes I forgot about soft validation. that works